### ISSUE TITLE
Hiding or categorizing commodities

### ISSUE THREAD

**Issue by Darksonn** [beancount/fava#1927]
My commodities view is currently not super useful because it shows a lot of different kinds of commodities mixed together. Would it be possible to mark some commodities as hidden so they don't show up? Or even better, add the option to categorize them and have the Commodities page let you pick the category you want to look at.

**Comment by mark-addams001881** [comment]
Hello, we tried to solve the issue.

This is what we did:

Modify the CommoditiesModule to include a category attribute for commodities and add a method to get commodities by category.

You can review changes in this commit: https://github.com/mark-addams001881/beancount-fava-1927/commit/b0697b06ba723e41383c997d1781152ff35f14cb.


> [!CAUTION]
> Disclaimer: The concept of solution was created by AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---

Latta AI seeks to solve problems in open source projects as part of its mission to support developers around the world. Learn more about our mission at https://latta.ai/ourmission . If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.

**Comment by yagebu** [comment]
Yeah, the commodities report is a mess. I never really use it (except for maybe the charts at the top, which are ok I guess). I guess any sort of change on the page would make an improvement...

The easiest to implement would probably be client-side filtering. Concrete ideas and PRs welcome.

What kind of categorisation were you thinking of? Having something like `asset_type: "stock"` and then grouping on prices involving stocks?
